app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperFileManager.kt

```diff
@@ -8,13 +8,20 @@ import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.mozilla.fenix.perf.runBlockingIncrement
 import java.io.File
 
 class WallpaperFileManager(
-    private val rootDirectory: File,
+    private val getRootDirectory: () -> File,
     coroutineDispatcher: CoroutineDispatcher = Dispatchers.IO
 ) {
     private val scope = CoroutineScope(coroutineDispatcher)
+    private val rootDirectory = runBlockingIncrement {
```

Comment:
Since we really want to temporarily suppress the strict mode violation, it would be better to do this explicitly: https://github.com/mozilla-mobile/fenix/blob/9ceb06e55c2b3a66cec6da55da51b90fe14e54d5/app/src/main/java/org/mozilla/fenix/experiments/NimbusSetup.kt#L51

The warning here is for an IO operation blocking the main thread. This isn't improved by blocking the main thread on an IO thread for that same operation. It just makes it riskier (additional thread needded) and slower because of the context switch.